receiver/awss3receiver/s3sqsreader.go

```diff
@@ -0,0 +1,217 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package awss3receiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awss3receiver"
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"go.uber.org/zap"
+)
+
+// S3 event notification structure from AWS
+// See: https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html
+
+// S3ObjectData represents an S3 object in the notification
+type S3ObjectData struct {
+	Key string `json:"key"`
+}
+
+// S3BucketData represents an S3 bucket in the notification
+type S3BucketData struct {
```

Comment:
Please move under internal or unexport those structs to limit the API exposed by this component.